### DISCUSSION TITLE
Client cert authentication instead of user / password

### DISCUSSION THREAD

**Discussion by stefan1983** [navidrome/navidrome#2364] [Ideas]
I would like to make my navidrome setup available from the internet and use client cert authentication instead of user / password login. Is that somehow possible with Navidrome?

**Comment by deluan** [comment]
Navidrome itself does not support such authentication schema, but maybe you can pair it with some authentication proxy that can do this. Then you can configure ND to use [ReverseProxyAuthentication](https://www.navidrome.org/docs/usage/security/#reverse-proxy-authentication).

Just keep in mind that while this may work with Navidrome's WebUI, it won't work with Subsonic apps, as the original [API](http://subsonic.org/pages/api.jsp#:~:text=with%20UTF%2D8.-,Authentication,-If%20your%20targeting) does not allow any authentication extensions.

**Comment by stefan1983** [comment]
What in general would be the best approach to secure your own local network then? Not use it at all? :)

Or is there any other best practice available?

**Reply by deluan** [reply]
Besides the [Security Considerations](https://www.navidrome.org/docs/usage/security) linked above, you could use a VPN (or a mesh VPN like Zerotier/Tailscale)
